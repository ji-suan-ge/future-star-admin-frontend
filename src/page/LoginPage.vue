<template>
  <el-container>
    <el-main>
      <el-card
        class="login-card"
        shadow="always">
        <el-row type="flex" justify="center" class="hint">
          <el-col :span="4">
            <img src="../assets/EdStarsLogo.jpg" class="logo"/>
          </el-col>
          <el-col :span="8" class="login-hint">
            管理人员登录平台
          </el-col>
        </el-row>
        <el-form label-width="90px">
          <el-form-item label="请输入账号:">
              <el-input type="text" autofocus="autofocus" class="input-text"/>
            </el-form-item>
          <el-form-item label="请输入密码:">
              <el-input type="password" class="input-text"/>
            </el-form-item>
          <el-form-item>
            <el-row type="flex" justify="center">
              <el-col :push="4">
                <el-button type="primary" @click="login()" class="login-button">登录</el-button>
              </el-col>
            </el-row>
          </el-form-item>
        </el-form>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'LoginPage',
  methods: {
    login: function () {
      this.$router.push('/home')
    }
  }
}
</script>

<style scoped type="type/css">
  .hint {
    margin-top: 5px;
  }

  .login-hint {
    margin-top: 15px;
    margin-left: 15px;
  }

  .logo {
    width: 66px;
    height: 60px;
  }

  .login-card {
    width: 38%;
    padding: 0;
    margin: 90px auto 80px;
  }

  .input-text {
    width: 300px;
    margin-left: 5px;
  }

  .login-button {
    width: 150px;
  }
</style>
