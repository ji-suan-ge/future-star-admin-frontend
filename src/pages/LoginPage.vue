<template>
    <div>
        login
    </div>
</template>

<script>
</script>

<style scoped type="type/css">
</style>
