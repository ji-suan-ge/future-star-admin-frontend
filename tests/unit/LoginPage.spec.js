import { mount } from '@vue/test-utils'
import LoginPage from '../../src/page/LoginPage'

describe('LoginPage', () => {
  const wrapper = mount(LoginPage)
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('按钮', async () => {
    const button = wrapper.find('el-button')
    await expect(button.text()).toBe('登录')
  })
})
