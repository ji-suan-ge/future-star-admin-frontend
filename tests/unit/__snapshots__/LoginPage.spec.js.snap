// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage 快照测试 1`] = `
<section class="el-container is-vertical">
  <header class="el-header" style="height: 60px;">
    <div class="el-row is-justify-center el-row--flex">
      <div class="el-image logo">
        <div class="el-image__placeholder"></div>
        <!---->
      </div>
    </div>
  </header>
  <div class="el-divider el-divider--horizontal">
    <!---->
  </div>
  <main class="el-main">
    <div class="el-row is-justify-center el-row--flex"><span class="platform-name">未来之星管理平台</span></div>
    <div class="el-row is-justify-center el-row--flex">
      <div class="el-card login-card is-always-shadow">
        <!---->
        <div class="el-card__body">
          <form class="el-form el-form--label-top">
            <div class="el-form-item"><label class="el-form-item__label">账号</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" autofocus="autofocus" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <div class="el-form-item"><label class="el-form-item__label">密码</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="password" autocomplete="off" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <div class="el-row">
              <div class="el-col el-col-17"><label class="el-checkbox"><span class="el-checkbox__input"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value="记住账号"></span><span class="el-checkbox__label">记住账号</span></label></div>
              <div class="el-col el-col-7"><a href="#/login" class="forget-password">
                  忘记密码？
                </a></div>
            </div>
            <div class="el-form-item">
              <!---->
              <div class="el-form-item__content">
                <div class="el-row"><button type="button" class="el-button login-button el-button--success">
                    <!---->
                    <!----><span>登录</span></button></div>
                <!---->
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </main>
  <div class="el-divider el-divider--horizontal">
    <!---->
  </div>
  <footer class="el-footer" style="height: 60px;">
    <div class="el-row is-justify-center el-row--flex">
      <div class="hint-footer el-col el-col-4">
        @copyright coo studio 2019
      </div>
    </div>
  </footer>
</section>
`;
